reference/src/main/java/com/google/android/fhir/reference/AddPatientViewModel.kt

```diff
@@ -50,7 +50,9 @@ class AddPatientViewModel(application: Application, private val state: SavedStat
    */
   fun savePatient(questionnaireResponse: QuestionnaireResponse) {
     viewModelScope.launch {
-      val patient = ResourceMapper.extract(questionnaireResource, questionnaireResponse) as Patient
+      val patient =
+        ResourceMapper.extract(questionnaireResource, questionnaireResponse).entry[0].resource as
```

Comment:
Could entry be an empty list? If it's possible, should you fail the save operation gracefully with error message rather than crashing the reference app?

Reply:
try something like this?

```suggestion
        val something = ResourceMapper.extract(questionnaireResource, questionnaireResponse).singleOrNull() ?: return
        val patient = something.resource as Patient
```

Reply:
Using `entryFirstRep` property of `Bundle` that promises to returns first entry. If entry list is empty, it will add an entry and then return the first. I am now checking for the type of the `entry.resource` and returning gracefully if its not Patient.